File: meditation.py
from flask import Blueprint, request, jsonify
from models import MeditationLog, DailyRoutine
from database import db
from services.gemini_service import get_meditation_suggestions

meditation_bp = Blueprint('meditation', __name__)

@meditation_bp.route('/api/meditation-log', methods=['POST'])
def log_meditation():
    data = request.json
    new_log = MeditationLog(user_id=data['user_id'], duration=data['duration'], type=data['type'])
    db.session.add(new_log)
    db.session.commit()
    return jsonify({'message': 'Meditation logged successfully', 'log_id': new_log.id}), 201

@meditation_bp.route('/api/meditation-suggestions', methods=['GET'])
def get_suggestions():
    user_id = request.args.get('user_id')
    # Fetch user's recent routines and meditation logs
    recent_routines = DailyRoutine.query.filter_by(user_id=user_id).order_by(DailyRoutine.date.desc()).limit(5).all()
    recent_logs = MeditationLog.query.filter_by(user_id=user_id).order_by(MeditationLog.date.desc()).limit(5).all()

    # Use Gemini API to get personalized suggestions
    suggestions = get_meditation_suggestions(recent_routines, recent_logs)
    return jsonify(suggestions)

File: meditation.js
document.addEventListener("DOMContentLoaded", () => {
  fetchMeditationSuggestions()
  document.getElementById("log-meditation-form").addEventListener("submit", logMeditation)
})

async function fetchMeditationSuggestions() {
  try {
    const response = await fetch("/api/meditation-suggestions?user_id=1") // Assuming user_id 1 for now
    const suggestions = await response.json()
    document.getElementById("meditation-suggestions").innerHTML = `<h3>Suggestions:</h3><p>${suggestions}</p>`
  } catch (error) {
    console.error("Error fetching meditation suggestions:", error)
  }
}

async function logMeditation(event) {
  event.preventDefault()
  const duration = document.getElementById("meditation-duration").value
  const type = document.getElementById("meditation-type").value

  try {
    const response = await fetch("/api/meditation-log", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ user_id: 1, duration, type }), // Assuming user_id 1 for now
    })
    if (response.ok) {
      alert("Meditation session logged successfully!")
      event.target.reset()
      fetchMeditationSuggestions()
    }
  } catch (error) {
    console.error("Error logging meditation:", error)
  }
}

